examples/viewcopy/viewcopy.cpp

```diff
@@ -96,7 +102,59 @@ void std_copy(
     std::copy(srcView.begin(), srcView.end(), dstView.begin());
 }
 
-void parallel_memcpy(std::byte* dst, const std::byte* src, std::size_t size, std::size_t numThreads = 1)
+// adapted from: https://stackoverflow.com/a/30386256/1034717
+void* memcpy_avx2(void* dst, const void* src, size_t n) noexcept
+{
+#define ALIGN(ptr, align) (((ptr) + (align) -1) & ~((align) -1))
+
+    char* d = static_cast<char*>(dst);
+    const char* s = static_cast<const char*>(src);
+
+    // fall back to memcpy() if dst and src are misaligned
+    if ((reinterpret_cast<uintptr_t>(d) & 31) != (reinterpret_cast<uintptr_t>(s) & 31))
+        return memcpy(d, s, n);
+
+    // align dst/src address multiple of 32
+    if (reinterpret_cast<uintptr_t>(d) & 31)
+    {
+        uintptr_t header_bytes = 32 - (reinterpret_cast<uintptr_t>(d) & 31);
+        assert(header_bytes < 32);
+
+        memcpy(d, s, std::min(header_bytes, n));
+
+        d = reinterpret_cast<char*>(ALIGN(reinterpret_cast<uintptr_t>(d), 32));
+        s = reinterpret_cast<char*>(ALIGN(reinterpret_cast<uintptr_t>(s), 32));
+        n -= std::min(header_bytes, n);
+    }
+
+    constexpr auto unrollFactor = 8;
```

Comment:
I did some experiments with the unroll factor:
with unroll 16:
```
memcpy_avx2 1.18689 s   7.37223GiB/s  
memcpy_avx2(p) 0.737673 s       11.8616GiB/s  
```
with unroll 8:
```
memcpy_avx2 1.18048 s   7.41226GiB/s      
memcpy_avx2(p) 0.701376 s       12.4755GiB/s   
```
with unroll 4:
```
memcpy_avx2 1.18842 s   7.36272GiB/s     
memcpy_avx2(p) 0.762377 s       11.4773GiB/s   
```
with unroll 2:
```
memcpy_avx2 1.21957 s   7.17467GiB/s
memcpy_avx2(p) 0.748464 s       11.6906GiB/s
```
handwritten version equivalent to unroll 2:
```
memcpy_avx2 1.20118 s   7.28448GiB/s 
memcpy_avx2(p) 0.736476 s       11.8809GiB/s        
```
It seems an unroll factor of 8 works quite ok. But all of this is probably very architecture specific.